File: swapXtemp.c
//Program to swap 2 numbers without temporary variable

#include<stdio.h>
void main()
{
    int a,b;
    printf("Enter two Integers: \n");
    scanf("%d",&a);
    scanf("%d",&b);
    printf("Numbers before swapping are: %d and %d\n",a,b);
    a=a+b;    //logic
    b=a-b;    //logic
    a=a-b;    //logic
    printf("Numbers after swapping are: %d and %d",a,b);
}

File: swapXtemp.py
# Program to swap two numbers without temporary variable

print('Enter two Integers:')
a=input()
b=input()
print('Numbers before swapping are: '+a+' and '+b)
a=int(a)+int(b)                                         #int convert need 
b=int(a)-int(b)
a=int(a)-int(b)
print('Numbers after swapping are: '+str(a)+' and '+str(b)) #str convert needed for concatnation